src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -454,11 +454,23 @@ bool LibraryCallKit::inline_vector_shuffle_iota() {
     // Wrap the indices greater than lane count.
     res = gvn().transform(VectorNode::make(Op_AndI, res, bcast_mod, num_elem, elem_bt));
   } else {
-    ConINode* pred_node = (ConINode*)gvn().makecon(TypeInt::make(BoolTest::ugt));
+    Node* mask = nullptr;
+    ConINode* pred_node = nullptr;
     Node * lane_cnt  = gvn().makecon(TypeInt::make(num_elem));
     Node * bcast_lane_cnt = gvn().transform(VectorNode::scalar2vector(lane_cnt, num_elem, type_bt));
-    Node* mask = gvn().transform(new VectorMaskCmpNode(BoolTest::ugt, bcast_lane_cnt, res, pred_node, vt));
 
+    if (Matcher::supports_unsigned_vector_comparison(num_elem, elem_bt)) {
+      pred_node = (ConINode*)gvn().makecon(TypeInt::make(BoolTest::ugt));
+      mask = gvn().transform(new VectorMaskCmpNode(BoolTest::ugt, bcast_lane_cnt, res, pred_node, vt));
+    } else {
+      // Currently it works well for vector_length <= 1024-bits.
+      // for vector_length > 1024, we don't support now
```

Comment:
Does it need any vector length check or block for vector length >1024?

Reply:
thank you for your reviw. It is a problem here. I will do that in next commit.

Reply:
As discussed with @nsjian , can we currently make it unsupported as well for 1024-bits? It doesn't need to change any IR if 1024-bits is not supported. We can revisit this issue in future for vector length >= 1024-bits